File: 0680-valid-palindrome-ii/0680-valid-palindrome-ii.java
class Solution {
    public boolean isPallindrome(String s, int left, int right, int count){
        if(count>1) return false;
        while(left<right) {
            if(s.charAt(left) == s.charAt(right)){
                left++; right--;
            }
            else 
                return isPallindrome(s, left+1, right, count+1) || isPallindrome(s, left, right-1, count+1);
        }
        return true;
    }
    public boolean validPalindrome(String s) {
        return isPallindrome(s, 0, s.length()-1, 0);
    }
}

File: 0680-valid-palindrome-ii/0680-valid-palindrome-ii.js
/**
 * @param {string} s
 * @return {boolean}
 */

var isPallindrome = function (s, i, j, count) {
    while(i<j){
        if(s[i] === s[j]){
            i++; j--;
        } else {
            if(count > 0){
                return false;
            } else if(i+1===j){
                return true;
            } else if( s[i+1] === s[j] && s[i] === s[j-1]) {
                return isPallindrome(s, i, j-1, count+1) || isPallindrome(s,i+1, j, count+1);
            } else if(s[i+1] === s[j]){
                i++; count++;
            } else if(s[i] === s[j-1]){
                j--; count++;
            } else {
                return false;
            }
        }
    }
    return true;
}
var validPalindrome = function(s) {
    return isPallindrome(s, 0, s.length-1, 0);
};